esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -4892,20 +4899,33 @@ static const char * checkGetStatsInput(const char * s)
 
 bool CWsWorkunitsEx::onWUGetStats(IEspContext &context, IEspWUGetStatsRequest &req, IEspWUGetStatsResponse &resp)
 {
+    //This function is deprecated for 7.x and will be removed shortly afterwards.
+    //Anything that cannot be implemented with the scope iterator is implemented as a post filter
     try
     {
-        const char* creatorType = checkGetStatsInput(req.getCreatorType());
-        const char* creator = checkGetStatsInput(req.getCreator());
-        const char* scopeType = checkGetStatsInput(req.getScopeType());
-        const char* scope = checkGetStatsInput(req.getScope());
-        const char* kind = checkGetStatsInput(req.getKind());
+        const char* creatorType = checkGetStatsNullInput(req.getCreatorType());
+        const char* creator = checkGetStatsNullInput(req.getCreator());
+        const char* scopeType = checkGetStatsNullInput(req.getScopeType());
+        const char* scope = checkGetStatsNullInput(req.getScope());
+        const char* kind = checkGetStatsNullInput(req.getKind());
         const char* measure = req.getMeasure();
 
-        StatisticsFilter filter(creatorType, creator, scopeType, scope, measure, kind);
+        WuScopeFilter filter;
+        StatisticsFilter statsFilter(creatorType, creator, "*", "*", "*", "*");
+        filter.addOutputProperties(PTstatistics);
+        if (scopeType)
+            filter.addScopeType(scopeType);
+        if (scope)
+            filter.addScope(scope);
+        if (kind)
+            filter.addOutputStatistic(kind);
+        if (measure)
+            filter.setMeasure(measure);
         if (!req.getMinScopeDepth_isNull() && !req.getMaxScopeDepth_isNull())
-            filter.setScopeDepth(req.getMinScopeDepth(), req.getMaxScopeDepth());
+            filter.setDepth(req.getMinScopeDepth(), req.getMaxScopeDepth());
         else if (!req.getMinScopeDepth_isNull())
-            filter.setScopeDepth(req.getMinScopeDepth());
+            filter.setDepth(req.getMinScopeDepth(), req.getMinScopeDepth());
```

Comment:
@ghalliday What if MinDepth is Null but maxDepth isn't?

Reply:
I suspect it will work as expected - getMinScopeDepth() will return 0.

Reply:
This service is also deprecated, so I'm not too converned about catching invalid input